python/ray/dag/compiled_dag_node.py

```diff
@@ -458,6 +527,22 @@ def _preprocess(self) -> None:
                 downstream_actor_handle = None
                 if isinstance(task.dag_node, ClassMethodNode):
                     downstream_actor_handle = task.dag_node._get_actor_handle()
+
+                # If the upstream node is an InputAttributeNode, treat the
+                # DAG's input node as the actual upstream node
+                if isinstance(upstream_node.dag_node, InputAttributeNode):
+                    assert (
+                        "dag_input_node"
+                        in upstream_node.dag_node.get_other_args_to_resolve()
+                    )
+                    dag_input_node_index = self.dag_node_to_idx[
+                        upstream_node.dag_node.get_other_args_to_resolve()[
```

Comment:
Suggest separating this out into a helper method on the `InputAttributeNode` class that returns the corresponding `InputNode`.

Reply:
The `InputNode` is already known so we don't need it from `InputAttributeNode`. This was a sanity check that can probably be removed. I cleaned up the code.